src/sqlfluff/core/linter/fix.py

```diff
@@ -118,16 +118,39 @@ def apply_fixes(
     dialect: "Dialect",
     rule_code: str,
     fixes: Dict[int, AnchorEditInfo],
+    fix_even_unparsable: bool = False,
```

Comment:
One thing that confuses me -- if a file has parsing errors and `fix_even_unparsable` is false, should this function even be called? There may be some nuance here I don't understand or have forgotten, but my initial instinct suggests there may be a more sweeping upstream fix possible. (By "sweeping", I mean at the file level rather than the parse subtree level.)

Reply:
Good point 🤔 .

When I check now, what you're suggesting does actually already exist, but further up in the linting process. In `lint_paths()` we only _persist_ the file if it's free of templating errors, which means you're kind of right.

Is that what you mean?

Reply:
@barrywhart - I think you're right that we need to rationalise how `fix_even_unparsable` is used across the codebase. On inspection it's pretty inconsistent, including how it interacts with `fix` and when it's read from config and when it's read from a keyword argument.

I think it's a bigger issue to fix, and I'm going to suggest doing that as a separate piece of work. I've make an issue to track it: #6264